feat: show unclaimed permits

FernandVEYRIER opened the pull request:
Resolves ubiquity/audit.ubq.fi#5 

<!--
- You must link the issue number e.g. "Resolves #1234"
- Please do not replace the keyword "Resolves" https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

pavlovcik commented:
> [`2a3fe2a`](https://0a9be4f9.pay-ubq-fi-bxp.pages.dev)

I broke the RPC thing and pushed to development (sorry about this!) For a quick review you might be able to pull from mine: https://github.com/ubiquity/pay.ubq.fi/pull/175

pavlovcik commented on static/scripts/audit-report/audit.ts:
```diff
@@ -306,7 +307,7 @@ class SmartQueue {
         elemList.push({
           id: git.issue_number,
-          tx: ether?.txHash || "N/A", // @TODO - handle this better
+          tx: ether?.txHash || TX_EMPTY_VALUE, // @TODO - handle this better
```
`null` is fine. And don't display `null` to the user.

FernandVEYRIER replied:
This one is not for display, it is the current default value when there is no `txHash` (just made it a variable to avoid typo mistakes)

pavlovcik commented on static/scripts/audit-report/helpers.ts:
```diff
@@ -31,8 +31,10 @@ const RPC_HEADER = {
+
 export function shortenTransactionHash(hash: string | undefined, length = 8): string {
-  if (!hash) return "";
+  if (!hash || hash === TX_EMPTY_VALUE) return "UNCLAIMED";
```
Don't display anything to the user.

rndquu reviewed (changes requested):
Here is the audit result for the https://github.com/ubiquity/business-development repo:
<img width="1512" alt="Screenshot 2024-02-22 at 13 01 48" src="https://github.com/ubiquity/pay.ubq.fi/assets/119500907/90ca679a-8c77-41df-aaed-58a8d4a3907f">

This [issue](https://github.com/ubiquity/business-development/issues/67) is missing in the final report. As far as I understand from the comments in that issue [those permits](https://github.com/ubiquity/business-development/issues/67#issuecomment-1941936242) were not claimed. Pls make sure that unclaimed permits are displayed.

FernandVEYRIER commented:
> Here is the audit result for the https://github.com/ubiquity/business-development repo: <img alt="Screenshot 2024-02-22 at 13 01 48" width="1512" src="https://private-user-images.githubusercontent.com/119500907/306952750-90ca679a-8c77-41df-aaed-58a8d4a3907f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2MDcxNzMsIm5iZiI6MTcwODYwNjg3MywicGF0aCI6Ii8xMTk1MDA5MDcvMzA2OTUyNzUwLTkwY2E2NzlhLThjNzctNDFkZi1hYWVkLTU4YThkNGEzOTA3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyMlQxMzAxMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wM2ZhMThjMmQzOTlkZWJjZWNhODY3Y2RjOGIyMGMxNGM3OGQ1ODc5ZWIzOTE0NDIxMjYzZDc0NmVjMzcxZDU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V2h0PSLVRjlVZHT7eHCg3RxN6IiSE5vmn2tG170E2Zg">
> 
> This [issue](https://github.com/ubiquity/business-development/issues/67) is missing in the final report. As far as I understand from the comments in that issue [those permits](https://github.com/ubiquity/business-development/issues/67#issuecomment-1941936242) were not claimed. Pls make sure that unclaimed permits are displayed.

Thanks for your input. I cannot test without the secret key and could only try on https://github.com/ubiquity/ubiquibot please let me know if the fix works.

I changed the logic so the items are added after all operations are complete instead of having them added after rpc + db fetch are complete, which should solve the issue. Added the corresponding reset as well.

pavlovcik commented:
>  I cannot test without the secret key 

What secret key

pavlovcik commented:
> > > I cannot test without the secret key
> > 
> > 
> > What secret key
> 
> Bot Wallet Address I meant, I think it is required to get the results from a certain repo?

Results are publicly posted on the blockchain and on GitHub comments. No private key required.

FernandVEYRIER commented:
> > > > I cannot test without the secret key
> > > 
> > > 
> > > What secret key
> > 
> > 
> > Bot Wallet Address I meant, I think it is required to get the results from a certain repo?
> 
> Results are publicly posted on the blockchain and on GitHub comments. No private key required.

You are correct, but since part of the results come from the blockchain, I do not know the Bot Wallet Address, but I realize that it is actually the address I see whenever transactions are made, so I could figure it out myself, my bad

pavlovcik commented:
I've got a few things on my plate and wont be able to handle this review. @rndquu hope that you can get this resolved. Also as a heads up @FernandVEYRIER as soon as this is merged in I would like to get this into another repo 
- #171

rndquu commented:
@FernandVEYRIER 

> I cannot test without the secret key

You can QA the audit page this way:
1. [Create](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) a new github personal access token
2. Use this quick import for testing:
```
 {
  "WALLET": "0xf87ca4583C792212e52720d127E7E0A38B818aD1",
  "REPO": "https://github.com/ubiquity/business-development",
  "PAT": "YOUR_PAT"
 }
```

Could you make sure that https://github.com/ubiquity/business-development/issues/67 is displayed in the audit results when running from the latest commit in this PR?

FernandVEYRIER commented:
> @FernandVEYRIER
> 
> > I cannot test without the secret key
> 
> You can QA the audit page this way:
> 
> 1. [Create](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) a new github personal access token
> 2. Use this quick import for testing:
> 
> ```
>  {
>   "WALLET": "0xf87ca4583C792212e52720d127E7E0A38B818aD1",
>   "REPO": "https://github.com/ubiquity/business-development",
>   "PAT": "YOUR_PAT"
>  }
> ```
> 
> Could you make sure that [ubiquity/business-development#67](https://github.com/ubiquity/business-development/issues/67) is displayed in the audit results when running from the latest commit in this PR?

@rndquu thank you. I was trying that way before, and I never get any result, which is why I started questioning about the WALLET value. Here is the result of the query
![image](https://github.com/ubiquity/pay.ubq.fi/assets/9807008/77568f69-897c-4905-87e6-f32aacf1d040)

I am not aware if there is any API limiter, or if I am missing some values in my env or something wrong with my setup in general. I am also using the database with id `wfzpewmlyiozupulbuur` which contains 0 permits for that repo. I will investigate in the meantime.

FernandVEYRIER commented:
@rndquu  After investigation, there seemed to be something wrong within the local database (browser's) that was saved, so after clearing the database the fetching works again so far. But now that we don't read the GitHub issues anymore, in order to be displayed the items have to be saved somewhere in the Supabase db, are you using another db than the one I have right now?

pavlovcik commented:
Yes we are in a transition period but the idea is to query all the necessary information from our database. The thing is, we are unsure how long it will take to finalize the database because the infrastructure is evolving so we need to change the database schema to accommodate. 

Fortunately with the plugin infrastructure we can isolate tables per plugin to make things more adaptable in our rapid r&d phase.

FernandVEYRIER commented:
Made significant changes to accommodate for the changes related to how the permits are fetched. Now the DB should be the only source of truth, holding transactions, amounts and users. This allows for a much better performance, since we do not need to fetch the transactions from the chains anymore, and can just rely on a one DB call to get all the needed data. We still need GitHub to fetch the user and issue details as they are not saved in the db.

pavlovcik commented:
> [`2a3fe2a`](https://0a9be4f9.pay-ubq-fi-bxp.pages.dev)
> [`05ba05a`](https://eafd8088.pay-ubq-fi-bxp.pages.dev)

I made some fixes to the continuous deployment- would you be able to merge in the latest so we can see your latest deployment automatically?

pavlovcik commented:
@rndquu can you handle this review

FernandVEYRIER commented:
@rndquu there is no record in the database for `business-development` repo, hance the empty result. Now all the data is pulled out from the database, from the transaction to wallets to permits, because I believe this is the way it should be done, or at least in a near future. The db should be holding these informations without us needing to parse each gnosis / ether transaction. You can give a try with `ubiquibot` repositories which I added for testing.

Eventually maybe we should consider a script that would help us populate the db with the previous transactions and users that are missing.

pavlovcik commented:
Those should still exist they are just private repositories.

rndquu commented:
> @rndquu there is no record in the database for `business-development` repo, hance the empty result. Now all the data is pulled out from the database, from the transaction to wallets to permits, because I believe this is the way it should be done, or at least in a near future. The db should be holding these informations without us needing to parse each gnosis / ether transaction. You can give a try with `ubiquibot` repositories which I added for testing.
> 
> Eventually maybe we should consider a script that would help us populate the db with the previous transactions and users that are missing.

So if there is no permit for an issue then such issue won't be displayed, right?

FernandVEYRIER commented:
> > @rndquu there is no record in the database for `business-development` repo, hance the empty result. Now all the data is pulled out from the database, from the transaction to wallets to permits, because I believe this is the way it should be done, or at least in a near future. The db should be holding these informations without us needing to parse each gnosis / ether transaction. You can give a try with `ubiquibot` repositories which I added for testing.
> > Eventually maybe we should consider a script that would help us populate the db with the previous transactions and users that are missing.
> 
> So if there is no permit for an issue then such issue won't be displayed, right?

Currently the fetching is done as such:
```TypeScript
    const { data } = await supabase
      .from("permits")
      .select("*, locations(*), users(*, wallets(*)), tokens(*)")
      .eq("locations.repository_id", gitData?.id)
      .not("locations", "is", null);
```
So indeed permits is the table that is queried. Would you rather have the issues queried and eventually link the permit if it exists?

rndquu commented:
> > > @rndquu there is no record in the database for `business-development` repo, hance the empty result. Now all the data is pulled out from the database, from the transaction to wallets to permits, because I believe this is the way it should be done, or at least in a near future. The db should be holding these informations without us needing to parse each gnosis / ether transaction. You can give a try with `ubiquibot` repositories which I added for testing.
> > > Eventually maybe we should consider a script that would help us populate the db with the previous transactions and users that are missing.
> > 
> > 
> > So if there is no permit for an issue then such issue won't be displayed, right?
> 
> Currently the fetching is done as such:
> 
> ```ts
>     const { data } = await supabase
>       .from("permits")
>       .select("*, locations(*), users(*, wallets(*)), tokens(*)")
>       .eq("locations.repository_id", gitData?.id)
>       .not("locations", "is", null);
> ```
> 
> So indeed permits is the table that is queried. Would you rather have the issues queried and eventually link the permit if it exists?

Yes, this was the original intention of the current issue

FernandVEYRIER commented:
![image](https://github.com/ubiquity/pay.ubq.fi/assets/9807008/2ac8ec56-d07f-4813-8f67-4b2b71c295ce)
This is the current DB schema. I thought the point of the audit page was to display all the current permits, regardless if they have been claimed or not. From my understanding after the comments, it should display all the issues and eventually if there is a permit, link it to that issue. Just posting this so we are all on the same page!

FernandVEYRIER commented:
Here is an example with an issue that doesn't have any permit claimed nor created:
![image](https://github.com/ubiquity/pay.ubq.fi/assets/9807008/2121a2fc-d16a-4be0-a073-314f5b870701)

Problem is this database doesn't not contain any relevant data, mostly tests done in pavlocik repositories. I do not know if we have some real data in another database for this, otherwise we should consider creating a script that would populate the db with relevant data, parsing the previous transactions and issues, let me know if this is needed.

pavlovcik commented:
Remarks on database schema:

Although I appreciate you taking the time to understand the relationships between each table, for the sake of modularity, I think it makes the most sense to have a separate table/db per plugin, which means that a full understanding of the relationships across all tables is no longer necessary. Instead the developer only needs to concern themselves with what plugin interacts with the data that's relevant to the task at hand. 

Locations table idea should be completely removed. The idea here was to keep track of where any activity happened for auditability purposes. The problem is that it introduced significant complexity and had minimal benefit. Instead we should just add a new column for the URL of the entity we are referring to (e.g. a GitHub comment, an issue, a user, a pull request etc) it is slightly space inefficient but makes working with the database significantly easier (and still allows for easy auditing)

The finance related stuff (permits, settlements) should be handled only by the permit generation plugin.

FernandVEYRIER commented:
> Remarks on database schema:
> 
> Although I appreciate you taking the time to understand the relationships between each table, for the sake of modularity, I think it makes the most sense to have a separate table/db per plugin, which means that a full understanding of the relationships across all tables is no longer necessary. Instead the developer only needs to concern themselves with what plugin interacts with the data that's relevant to the task at hand.
> 
> Locations table idea should be completely removed. The idea here was to keep track of where any activity happened for auditability purposes. The problem is that it introduced significant complexity and had minimal benefit. Instead we should just add a new column on every table for the URL of the entity we are referring to (e.g. a GitHub comment, an issue, a user, a pull request etc) it is slightly space inefficient but makes working with the database significantly easier (and still allows for easy auditing)
> 
> The finance related stuff (permits, settlements, credits, debits) should be handled only by the permit generation plugin.
> 
> And so on, the other tables should have specific plugins they are associated with.
> 
> The only exception I potentially see: logs are nice to have for the kernel actually, unless cloudflare has a built in solution that is good (we built this cause netlify wasn't great) if cloudflare has a good and free solution then perhaps we should migrate to that.

The problem working with urls for audit is that it implies parsing the page like now for comments, urls, prs and so on, which is very inefficient and unreliable (if tomorrow GitHub decides to change one div name it breaks everything) so I think the database on that regard was beneficial. It speeds up the process significantly and allows for scaling because the more issues there are the slower it would be parsing them. I agree this is a case by case thing, but for example for this repo using a db instead of cross referencing gnosis / ether / pull-requests made the code significantly easier to read and maintain. Maybe we should have a separated module only handling GitHub events with hooks and keeping records in a database? It might be a viable solution. Let me know, otherwise we can abandon this PR I guess.

pavlovcik commented:
> > Remarks on database schema:
> 
> > 
> 
> > Although I appreciate you taking the time to understand the relationships between each table, for the sake of modularity, I think it makes the most sense to have a separate table/db per plugin, which means that a full understanding of the relationships across all tables is no longer necessary. Instead the developer only needs to concern themselves with what plugin interacts with the data that's relevant to the task at hand.
> 
> > 
> 
> > Locations table idea should be completely removed. The idea here was to keep track of where any activity happened for auditability purposes. The problem is that it introduced significant complexity and had minimal benefit. Instead we should just add a new column on every table for the URL of the entity we are referring to (e.g. a GitHub comment, an issue, a user, a pull request etc) it is slightly space inefficient but makes working with the database significantly easier (and still allows for easy auditing)
> 
> > 
> 
> > The finance related stuff (permits, settlements, credits, debits) should be handled only by the permit generation plugin.
> 
> > 
> 
> > And so on, the other tables should have specific plugins they are associated with.
> 
> > 
> 
> > The only exception I potentially see: logs are nice to have for the kernel actually, unless cloudflare has a built in solution that is good (we built this cause netlify wasn't great) if cloudflare has a good and free solution then perhaps we should migrate to that.
> 
> 
> 
> The problem working with urls for audit is that it implies parsing the page like now for comments, urls, prs and so on, which is very inefficient and unreliable (if tomorrow GitHub decides to change one div name it breaks everything) so I think the database on that regard was beneficial. It speeds up the process significantly and allows for scaling because the more issues there are the slower it would be parsing them. I agree this is a case by case thing, but for example for this repo using a db instead of cross referencing gnosis / ether / pull-requests made the code significantly easier to read and maintain. Maybe we should have a separated module only handling GitHub events with hooks and keeping records in a database? It might be a viable solution. Let me know, otherwise we can abandon this PR I guess.

The GitHub issues REST API is actually designed for the following parameters:

Organization name, repository name, issue/pull request number. 

These can easily be extracted from the URL. 

---

I assumed that if I had the comment ID, that I could directly look up the comment; but we would have to use graphql for all that. I would much, much rather we use the normal REST API for all of our infrastructure because it's significantly easier with its built in methods, and comes with type safety.

FernandVEYRIER commented:
> > > Remarks on database schema:
> > 
> > 
> > > 
> > 
> > 
> > > Although I appreciate you taking the time to understand the relationships between each table, for the sake of modularity, I think it makes the most sense to have a separate table/db per plugin, which means that a full understanding of the relationships across all tables is no longer necessary. Instead the developer only needs to concern themselves with what plugin interacts with the data that's relevant to the task at hand.
> > 
> > 
> > > 
> > 
> > 
> > > Locations table idea should be completely removed. The idea here was to keep track of where any activity happened for auditability purposes. The problem is that it introduced significant complexity and had minimal benefit. Instead we should just add a new column on every table for the URL of the entity we are referring to (e.g. a GitHub comment, an issue, a user, a pull request etc) it is slightly space inefficient but makes working with the database significantly easier (and still allows for easy auditing)
> > 
> > 
> > > 
> > 
> > 
> > > The finance related stuff (permits, settlements, credits, debits) should be handled only by the permit generation plugin.
> > 
> > 
> > > 
> > 
> > 
> > > And so on, the other tables should have specific plugins they are associated with.
> > 
> > 
> > > 
> > 
> > 
> > > The only exception I potentially see: logs are nice to have for the kernel actually, unless cloudflare has a built in solution that is good (we built this cause netlify wasn't great) if cloudflare has a good and free solution then perhaps we should migrate to that.
> > 
> > 
> > The problem working with urls for audit is that it implies parsing the page like now for comments, urls, prs and so on, which is very inefficient and unreliable (if tomorrow GitHub decides to change one div name it breaks everything) so I think the database on that regard was beneficial. It speeds up the process significantly and allows for scaling because the more issues there are the slower it would be parsing them. I agree this is a case by case thing, but for example for this repo using a db instead of cross referencing gnosis / ether / pull-requests made the code significantly easier to read and maintain. Maybe we should have a separated module only handling GitHub events with hooks and keeping records in a database? It might be a viable solution. Let me know, otherwise we can abandon this PR I guess.
> 
> The GitHub issues REST API is actually designed for the following parameters:
> 
> Organization name, repository name, issue/pull request number.
> 
> These can easily be extracted from the URL.
> 
> I assumed that if I had the comment ID, that I could directly look up the comment; but we would have to use graphql for all that. I would much, much rather we use the normal REST API for all of our infrastructure because it's significantly easier with its built in methods, and comes with type safety.
> 
> The alternative is that we have the comment ID embedded as the # parameter in the URL, then we load that issue/pr, and we can find the comment ID from there.

As a personal taste, GraphQl is usually way easier to use and is even more type safe than REST APIs, since types are dynamically generated from passed arguments where in REST you can send anything. But not using the db, we would have to also call again the ether / gnosis network to check for the status of the claim isn't it?

pavlovcik commented:
> > > > Remarks on database schema:
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > Although I appreciate you taking the time to understand the relationships between each table, for the sake of modularity, I think it makes the most sense to have a separate table/db per plugin, which means that a full understanding of the relationships across all tables is no longer necessary. Instead the developer only needs to concern themselves with what plugin interacts with the data that's relevant to the task at hand.
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > Locations table idea should be completely removed. The idea here was to keep track of where any activity happened for auditability purposes. The problem is that it introduced significant complexity and had minimal benefit. Instead we should just add a new column on every table for the URL of the entity we are referring to (e.g. a GitHub comment, an issue, a user, a pull request etc) it is slightly space inefficient but makes working with the database significantly easier (and still allows for easy auditing)
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > The finance related stuff (permits, settlements, credits, debits) should be handled only by the permit generation plugin.
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > And so on, the other tables should have specific plugins they are associated with.
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > The only exception I potentially see: logs are nice to have for the kernel actually, unless cloudflare has a built in solution that is good (we built this cause netlify wasn't great) if cloudflare has a good and free solution then perhaps we should migrate to that.
> 
> > > 
> 
> > > 
> 
> > > The problem working with urls for audit is that it implies parsing the page like now for comments, urls, prs and so on, which is very inefficient and unreliable (if tomorrow GitHub decides to change one div name it breaks everything) so I think the database on that regard was beneficial. It speeds up the process significantly and allows for scaling because the more issues there are the slower it would be parsing them. I agree this is a case by case thing, but for example for this repo using a db instead of cross referencing gnosis / ether / pull-requests made the code significantly easier to read and maintain. Maybe we should have a separated module only handling GitHub events with hooks and keeping records in a database? It might be a viable solution. Let me know, otherwise we can abandon this PR I guess.
> 
> > 
> 
> > The GitHub issues REST API is actually designed for the following parameters:
> 
> > 
> 
> > Organization name, repository name, issue/pull request number.
> 
> > 
> 
> > These can easily be extracted from the URL.
> 
> > 
> 
> > I assumed that if I had the comment ID, that I could directly look up the comment; but we would have to use graphql for all that. I would much, much rather we use the normal REST API for all of our infrastructure because it's significantly easier with its built in methods, and comes with type safety.
> 
> > 
> 
> > The alternative is that we have the comment ID embedded as the # parameter in the URL, then we load that issue/pr, and we can find the comment ID from there.
> 
> 
> 
> As a personal taste, GraphQl is usually way easier to use and is even more type safe than REST APIs, since types are dynamically generated from passed arguments where in REST you can send anything.

I mean REST using the @octokit/rest module. We manipulate the REST API with the provided type safe methods. I am not 100% sure about the graphql sdk but given that we pass in huge arbitrary strings it seems unlikely to me that it supports type safety. 

> But not using the db, we would have to also call again the ether / gnosis network to check for the status of the claim isn't it?

Yes but in my earlier research from a few days ago, in the context of pay.ubq.fi rewards interface: given the amount of functional RPC endpoints we have available to us, and given that most people claim within 12 hours of posting; we can feasibly find any specific claim status almost "realtime" at page load

I'm mobile so it's not too easy for me to find the original conversation on this.

pavlovcik commented:
Found it https://github.com/ubiquity/pay.ubq.fi/issues/138#issuecomment-1962326614

FernandVEYRIER commented:
@pavlovcik You can simply use codegen-graphql on GitHub endpoints to automatically generate all the TypeScript interfaces 😄
To make it real time I thought we could subscrible to GitHub hooks on pull request updates, and write the db on permit claims directly so everything is real time. Problem was with the previous version of this package, on busy repositories the audit took me around 3 / 4 minutes to fetch everything.

pavlovcik commented:
Because we don't have any test data in the database, its hard to end-to-end test this. Based on the result from pay.ubq.fi at least it shows me that it renders details. 

I think we can merge in and follow up here if theres issues down the line.
 
<img width="603" alt="Screenshot 2024-02-29 at 15 01 56" src="https://github.com/ubiquity/pay.ubq.fi/assets/4975670/b9c8f619-a5c3-452d-9231-7e2764e3bd1d">

FernandVEYRIER commented:
> Because we don't have any test data in the database, its hard to end-to-end test this. Based on the result from pay.ubq.fi at least it shows me that it renders details.
> 
> I think we can merge in and follow up here if theres issues down the line.
> 
> <img alt="Screenshot 2024-02-29 at 15 01 56" width="603" src="https://private-user-images.githubusercontent.com/4975670/308812347-b9c8f619-a5c3-452d-9231-7e2764e3bd1d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxODg1MjEsIm5iZiI6MTcwOTE4ODIyMSwicGF0aCI6Ii80OTc1NjcwLzMwODgxMjM0Ny1iOWM4ZjYxOS1hNWMzLTQ1MmQtOTIzMS03ZTI3NjRlM2JkMWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjlUMDYzMDIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTVjOWY2NGNhYzU2MWJiNDNlYjRiNmMyYTQ2MjFkYzI2OTk2MmVhZjEzNjhlNDAyMjk5YzM5MWIyZGViZmI2NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.3nkW9OoNgUB3boq0X4OS8yRPDZCviApw35Sokr-jm2g">
> Although to be honest I'm confused why this is here. Is it just a test entry? I can't find the permit generated.
> 
> #174
> 
> ```json
>  {
>   "WALLET": "0x44ca15db101fd1c194467db6af0c67c6bbf4ab51",
>   "REPO": "https://github.com/ubiquity/pay.ubq.fi",
>   "PAT": ""
>  }
> ```

@pavlovcik yes I manually entered this one for the sake of testing. I can create a script to populate the db with real data if needed

pavlovcik commented:
No worries on the script. We can test the rest in production. Thank you

rndquu commented:
To sum up: we're good to show issues only with claimed permits because it's easier and faster to read from DB instead of parsing github or onchain data

FernandVEYRIER commented:
> To sum up: we're good to show issues only with claimed permits because it's easier and faster to read from DB instead of parsing github or onchain data

Issues without permit should be also showing with the latest fix, as long as there is a record of that issue in the db

rndquu commented:
> > To sum up: we're good to show issues only with claimed permits because it's easier and faster to read from DB instead of parsing github or onchain data
> 
> Issues without permit should be also showing with the latest fix, as long as there is a record of that issue in the db

But there is no "issues" table in the DB, where do we get issues related info?

pavlovcik commented:
I really want to drop the `locations` table because its a liability dealing with all the foreign keys I made to it.

FernandVEYRIER commented:
> I really want to drop the `locations` table because its a liability dealing with all the foreign keys I made to it.

Might want to rethink the structure of the database? We should think what would be the most efficient way to sync the data with the GitHub and the transactions.

pavlovcik commented:
Definitely URL. At minimum it allows for instant auditability just by clicking the link on the Supabase UI.

FernandVEYRIER commented:
> Definitely URL. At minimum it allows for instant auditability just by clicking the link on the Supabase UI.

`locations` have a `node_url` containing the url of the issues currently

rndquu reviewed (approved):
To sum up:
1. We're retrieving issues from DB from the `locations` table
2. In the future the `locations` table will be removed so the audit page will have to be refactored to read issues from some other table

@FernandVEYRIER Could you reopen this PR towards the https://github.com/ubiquity/audit.ubq.fi repo?

pavlovcik commented:
> > Definitely URL. At minimum it allows for instant auditability just by clicking the link on the Supabase UI.
> 
> 
> 
> `locations` have a `node_url` containing the url of the issues currently

I know, I implemented all of the current database schema. 

We instead should have (probably redundant) URL columns on every table that links the GitHub entity at hand for the row data.

FernandVEYRIER commented:
@rndquu Reopened the PR at https://github.com/ubiquity/audit.ubq.fi/pull/2

pavlovcik commented:
> @rndquu Reopened the PR at [ubiquity/audit.ubq.fi#2](https://github.com/ubiquity/audit.ubq.fi/pull/2)

You should close this pull request if its no longer necessary.

FernandVEYRIER commented:
@pavlovcik I resolved the merge conflict in case you decide to merge it, everything should be solved at that point. The other PR will be awaiting for reviews in the meantime

rndquu commented:
Closing in favour of https://github.com/ubiquity/audit.ubq.fi/pull/2
